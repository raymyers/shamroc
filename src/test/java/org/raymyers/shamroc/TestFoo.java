package org.raymyers.shamroc;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;


import static org.junit.jupiter.api.Assertions.assertTrue;

public class TestFoo {
    @Test
    void basicParse()
    {
        SexprLoader subject = new SexprLoader("""
        (if a (if b c))
        """);
        assertTrue(subject.valid());
        Approvals.verify(subject.getStringTree());
    }
}
