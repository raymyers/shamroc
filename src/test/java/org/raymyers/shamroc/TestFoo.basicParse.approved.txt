(sexpr (item (list_ ( (item (atom if)) (item (atom a)) (item (list_ ( (item (atom if)) (item (atom b)) (item (atom c)) ))) ))) <EOF>)